examples/gallery/embellishments/rose.py

```diff
@@ -0,0 +1,42 @@
+"""
+Rose diagram
+------------
+
+The :meth:`pygmt.Figure.rose` method can plot windrose diagrams or polar histograms.
+"""
+
+import pygmt
+
+fig = pygmt.Figure()
+
+fig.rose(
+    # use the remote file fractures_06.txt which contains a compilation
+    # of fracture lengths and directions as digitized from geological maps
+    data="@fractures_06.txt",
```

Comment:
Might need to add this to the cache workflow at https://github.com/GenericMappingTools/pygmt/blob/37ce2f703daef97809ed2c2a25050e55dd93847d/.github/workflows/cache_data.yaml#L45-L47

Reply:
Opened a PR for this @weiji14.

Reply:
I think we should add a new function to load this data (see https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/datasets/tutorial.py for similar functions) instead of using GMT's remote file (`@`) mechanism.

Reply:
Is this because including a function to load the data is more pythonic? Are there any other reasons for avoiding the remote file mechanism in PyGMT?

Reply:
I think the reasons are:

1. More Pythonic.
2. Any remote files may be updated, removed or renamed. Updating a function is much easier than searching for `@fractures_06.txt` 
3. We can explain what the data is in the function docstring, otherwise we have to explain the data in every tutorial and examples that use the data.

Reply:
Thanks for sharing the explanation - all good points

Reply:
Sounds good to me 👍. I will close #1097 and open a new PR to add a new function for loading the data.